src/runtime.jl

```diff
@@ -0,0 +1,87 @@
+# Things that are useful to know across platforms
+if Sys.iswindows()
+    const LIBPATH_env = "PATH"
+    const LIBPATH_default = ""
+    const pathsep = ';'
+elseif Sys.isapple()
+    const LIBPATH_env = "DYLD_FALLBACK_LIBRARY_PATH"
+    const LIBPATH_default = "~/lib:/usr/local/lib:/lib:/usr/lib"
+    const pathsep = ':'
+else
+    const LIBPATH_env = "LD_LIBRARY_PATH"
+    const LIBPATH_default = ""
+    const pathsep = ':'
+end
+
+function adjust_ENV!(env::Dict, PATH::String, LIBPATH::String, adjust_PATH::Bool, adjust_LIBPATH::Bool)
+    if adjust_PATH
+        if !isempty(get(env, "PATH", ""))
+            env["PATH"] = string(PATH, pathsep, env["PATH"])
+        else
+            env["PATH"] = PATH
+        end
+    end
+    if adjust_LIBPATH
+        LIBPATH_base = get(env, LIBPATH_env, expanduser(LIBPATH_default))
+        if !isempty(LIBPATH_base)
+            env[LIBPATH_env] = string(LIBPATH, pathsep, LIBPATH_base)
+        else
+            env[LIBPATH_env] = LIBPATH
+        end
+    end
+    return env
+end
+
+function withenv_executable_wrapper(f::Function,
+                                    executable_path::String,
+                                    PATH::String,
+                                    LIBPATH::String,
+                                    adjust_PATH::Bool,
+                                    adjust_LIBPATH::Bool)
+    env = Dict{String,String}(
+        "PATH" => get(ENV, "PATH", ""),
+        LIBPATH_env => get(ENV, LIBPATH_env, ""),
+    )
+    env = adjust_ENV!(env, PATH, LIBPATH, adjust_PATH, adjust_LIBPATH)
+    withenv(env...) do
+        f(executable_path)
+    end
+end
+
+function dev_jll(src_name)
```

Comment:
In https://github.com/JuliaPackaging/BinaryBuilder.jl/pull/880#discussion_r483638009 Kristoffer suggested to move this function to some other package

Reply:
By putting it here, we're not baking it into each JLL, it will only be loaded at runtime (it is loaded as part of the `.ji` file for `JLLWrappers`, but that's sufficiently low-impact I think it's okay.  We can move it out later if we want).